### DISCUSSION TITLE
Occasional error: Error in try(self$crew$finalize()) : attempt to apply non-function

### DISCUSSION THREAD

**Discussion by Aariq** [mschubert/clustermq#286] [Q&A] [answered]
I'm using `targets` with `tar_make_clustermq()` to run targets on my university HPC.  Ever since they enabled 2FA (I think) I _occasionally_ get this error:

```
Connecting hpg via SSH ...
The following have been reloaded with a version change:
  1) R/4.1 => R/4.0
Error in try(self$crew$finalize()) : attempt to apply non-function
✓ skip pipeline
Error in initialize(...) : 
  Remote R process did not respond after 5000 seconds. Check your SSH server log.
```

It is nearly instantaneous, not 5000 seconds.  There's nothing unusual on the SSH log as far as I can tell, and if I just run `tar_make_clustermq()` again it usually works.  Any ideas on what this is and how I could go about doing a better job troubleshooting it?  I'm hoping maybe someone can at least tell me if the error is coming from `clustermq` or something else.

**Comment by mschubert** [comment, marked as answer]
There's a bug in this error message: It's 5 seconds, not 5000. Could that cover your "nearly instantaneous" return?

In any case, setting `options(clustermq.ssh.timeout=30)` is probably a good choice with 2FA.

**Reply by Aariq** [reply]
Ah yes, 5 seconds is reasonable!  I'll try adding a longer timeout.
